lib/ansible/plugins/filter/core.py

```diff
@@ -294,25 +293,93 @@ def mandatory(a):
 
 
 def combine(*terms, **kwargs):
-    recursive = kwargs.get('recursive', False)
-    if len(kwargs) > 1 or (len(kwargs) == 1 and 'recursive' not in kwargs):
-        raise AnsibleFilterError("'recursive' is the only valid keyword argument")
-
-    dicts = []
-    for t in terms:
-        if isinstance(t, MutableMapping):
-            recursive_check_defined(t)
-            dicts.append(t)
-        elif isinstance(t, list):
-            recursive_check_defined(t)
-            dicts.append(combine(*t, **kwargs))
-        else:
-            raise AnsibleFilterError("|combine expects dictionaries, got " + repr(t))
-
-    if recursive:
-        return reduce(merge_hash, dicts)
-    else:
-        return dict(itertools.chain(*map(iteritems, dicts)))
+    recursive = kwargs.pop('recursive', False)
+    list_merge = kwargs.pop('list_merge', 'keep')
```

Comment:
Is `list_merge` new argument ? If yes, then it would be nice to add documentation, examples, changelog entry for this change.

Reply:
Also, test case for this change.

Reply:
@Akasurde yes `list_merge` is a new argument.

This feature (fine handling of list merging during combine) is still in discussed. we will talk about it next Tuesday. That's why I didn't bother to write documentation nor tests.
I will do it after the implementation get validated.

> README: will come if this implementation is validated.

Reply:
OK, I didn't know this.